### DISCUSSION TITLE
Is there a good/better way to bridge bibtex/Zotero and papis?

### DISCUSSION THREAD

**Discussion by ThSGM** [papis/papis#958] [General]
In essence, Zotero provide a good GUI for handling paper PDF management. However, what it does not do well is it does not provide the kind of CLI power that papis provides. The question is how do you use Zotero in conjunction with Papis.

There is [one guide here](https://nicolasshu.com/zotero_and_papis.html) but it is clunky. The author is essentially suggesting for every single paper you own to be included within a titled folder where both the PDF can be housed, along with the info.yaml file that papis needs.

But this will absolutely wreak your PDF/folder organisation. For example, I rely on being able to quickly find PDFs on my computer via a fuzzy-finder like Alfred. This would absolutely pollute your search tree, unless you specifically restricted it to only PDFs. Is there an alternative route to using papis in conjunction with Zotero and its PDF management?

**Comment by alexfikl** [comment]
Hi! Sorry for the late reply.. I mostly read this and had no ideas for useful suggestions, so maybe someone else will show up with some better ideas.

First idea that comes to mind, if you like Zotero, is to just sidestep Papis completely here. Zotero stores all its metadata in a very nice sqlite database and it's very easy to access it and modify it. One such way could be using https://github.com/urschrei/pyzotero. It might take some work, but it seems doable to build a basic cli using `pyzotero`. Someone seems to have tried a long time ago: https://github.com/jbaiter/zotero-cli.

> But this will absolutely wreak your PDF/folder organisation. For example, I rely on being able to quickly find PDFs on my computer via a fuzzy-finder like Alfred. This would absolutely pollute your search tree, unless you specifically restricted it to only PDFs.

I'm not quite sure what you mean here (both from the Papis and Zotero sides). For Papis, individual documents are kept in separate folders that contain an `info.yml` file and any other PDFs / nodes / whatever. However, you can have a structure such as
```
Documents
- > MyPDFs
- > PapisLibrary
	- > doc-1
		- > info.yml
		- > file.pdf (symlink to some file in MyPDFs)
	- > doc-2
	- ... 
```
i.e. 
- papis uses its own directory structure for documents
- all your PDFs are stored in a single folder somewhere that's easily searchable for you. 
 
Using symlinks can be done when adding documents using `papis add --link [...]` or when importing from Zotero using `papis zotero import --link [...]`. Zotero also supports symlinks in its library, so there shouldn't be anything stopping you from only having one central location for your PDFs between the two.

However, keeping Papis and Zotero in sync in a way that they work well together doesn't seem easy at all..

> Is there an alternative route to using papis in conjunction with Zotero and its PDF management?

As far as I know, there isn't anything ready-made for this at the moment, but there are probably ways to hack around it :grin: If you need any help getting something from papis, let us know!

**Comment by alejandrogallo** [comment]
Unfortunately I'm not aware of another strategy to use papis alongside
Zotero. However, have you checked the web application that papis
includes with the distribution?, maybe this might work for you and you
might want to implement some other features that you're missing from
Zotero.

-- 
Alejandro Gallo
